### ISSUE TITLE
Updating P_pred

### ISSUE THREAD

**Issue by xiaojiacong2025-dot** [hi-zhenyu/PVC#2]
[main.py](https://github.com/hi-zhenyu/PVC/blob/0608f31adb04c69d9a020298e877d57c397f7e2c/main.py#L135)
From the current training code, it seems that P_pred is passed into the loss as a fixed input:

loss = criterion(X_list, ae_encoded, ae_decoded, P_pred)

and in PVC_Loss, it is only used in:

Loss2 = mse(ae_encoded_list[0], torch.mm(P, ae_encoded_list[1]))

So it looks like P_pred is not updated inside the training loop, and the optimizer only updates the network parameters.
